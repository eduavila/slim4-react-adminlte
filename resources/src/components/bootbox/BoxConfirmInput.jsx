import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-bootstrap/lib/Modal';
import Button from 'react-bootstrap/lib/Button';

class BoxConfirmInput extends Component{
    constructor(props){
        super(props)

        this.state = {
            result: ''
        };
    }

    onChange = (e)=>{
        this.setState({
            result: e.target.value
        });
    }

    render(){
        return (
            <Modal bsSize='lg' show={ this.props.show } animation={true} onHide={(e) => this.props.onClose({ isCancel:false }) }>
                <Modal.Header closeButton>
                     <Modal.Title>{this.props.message}</Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    <input type="text" autoComplete="off" className="form-control" value={this.state.result} onChange={this.onChange}/>
                </Modal.Body>
                <Modal.Footer>
                    <Button className="btn btn-secondary" onClick={() => this.props.onClose({ isCancel:true })}>
                        Cancelar
                    </Button>
                    <Button className="btn btn-primary" onClick={() => this.props.onConfirm(this.state.result)} disabled={this.state.result.length < 5}>
                        Confirmar
                    </Button>
                </Modal.Footer>
            </Modal>  
        );
    }
}

BoxConfirmInput.propTypes = {
  show: PropTypes.bool.isRequired,
  onConfirm: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired,
  message: PropTypes.string,
};

export default BoxConfirmInput;