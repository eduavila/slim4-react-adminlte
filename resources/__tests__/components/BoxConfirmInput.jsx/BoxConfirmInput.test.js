import React from 'react';
import { shallow,mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import BoxConfirmInput from '../../../src/components/bootbox/BoxConfirmInput';

describe('<BoxConfirmInput />', () => {
    describe('render()', () => {
        test('renders the component', () => {
            const wrapper = shallow(<BoxConfirmInput show={true} onClose={()=>{ }} onConfirm={()=>{ }} />);
        
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        test('Click botao confirmar', () => {
            const mockOnClick = jest.fn();
            const wrapper = mount(<BoxConfirmInput show={true} onClose={()=>{ }} onConfirm={mockOnClick} />);

            wrapper.find('input').first().simulate('change', { target: { value: 'Texto de confirmação .' } })
            wrapper.find('.btn-primary').first().simulate('click');

            expect(mockOnClick.mock.calls.length).toEqual(1);
        });

        test('Click botao cancelar', () => {
            const mockOnClick = jest.fn();
            const wrapper = mount(<BoxConfirmInput show={true} onClose={mockOnClick} onConfirm={mockOnClick} />);

            wrapper.find('.btn-secondary').first().simulate('click');

            expect(mockOnClick.mock.calls.length).toEqual(1);
            expect(mockOnClick.mock.calls[0][0]).toEqual({ isCancel:true });
        });

        test('Click botao cancelar', () => {
            const mockOnClick = jest.fn();
            const wrapper = mount(<BoxConfirmInput show={true} onClose={mockOnClick} onConfirm={mockOnClick} />);

            wrapper.find('button.close').first().simulate('click');

            expect(mockOnClick.mock.calls.length).toEqual(1);
            expect(mockOnClick.mock.calls[0][0]).toEqual({ isCancel:false });
        });
    });
});