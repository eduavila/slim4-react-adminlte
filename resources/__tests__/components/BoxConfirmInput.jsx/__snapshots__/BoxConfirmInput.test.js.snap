// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoxConfirmInput /> render() renders the component 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  bsSize="lg"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    />
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <input
      autoComplete="off"
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn btn-secondary"
      disabled={false}
      onClick={[Function]}
    >
      Cancelar
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="btn btn-primary"
      disabled={true}
      onClick={[Function]}
    >
      Confirmar
    </Button>
  </ModalFooter>
</Modal>
`;
